CycloneDX/Services/ProjectAssetsFileService.cs

```diff
@@ -95,51 +92,13 @@ public HashSet<NugetPackage> GetNugetPackages(string projectFilePath, string pro
 
             return packages;
         }
-        public bool SetIsDirectReference(string packageName, string targetRuntime, JsonElement jsonContent)
+        public bool SetIsDirectReference(LibraryDependency dependency)
         {
-            string framework = TargetFrameworkToAlias(targetRuntime);
-            JsonElement packageProperties;
-            if (jsonContent.GetProperty(framework).GetProperty("dependencies").TryGetProperty(packageName, out packageProperties))
-            {
-                // every direct reference has target property
-                return packageProperties.TryGetProperty("target", out _);
-            }
-            return false;
+            return dependency?.ReferenceType == LibraryDependencyReferenceType.Direct;
```

Comment:
I made sure by checking the implementation that this is indeed how the target is mapped to. It also has the benefit that it is already mapped onto exactly what we expect (and catches some other side-cases not caught by the existing code).